mne/encoding/model.py

```diff
@@ -0,0 +1,230 @@
+        An optional array of indices for manually specifying which samples
+        to keep during prediction. Defaults to None.
+    mask_condition : 'all', 'any'
+        Whether only 1 `mask_val` will trigger removal of a row, or if
+        all values of the row must be `mask_val` for removal.
+    """
+    def __init__(self, estimator, mask_val=None, ixs=None, ixs_pred=None,
+                 mask_condition='all'):
+        if all(ii is not None for ii in [mask_val, ixs]):
+            raise ValueError('Supply one of mask_val | ixs, or both')
+        if ixs is not None:
+            ixs = np.asarray(ixs).astype(int)
+        self.ixs = ixs
+        if ixs_pred is not None:
+            ixs_pred = np.asarray(ixs_pred).astype(int)
+        self.ixs_pred = ixs_pred
+
+        self.mask_val = mask_val
+        if mask_condition not in ['all', 'any']:
+            raise ValueError('condition must be one of "all" or "any"')
+        self.mask_condition = mask_condition
+        self.est = _check_estimator(estimator)
+
+    def fit(self, X, y):
+        """Remove datapoints then fit the estimator.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Rows will be removed.
+        y : array, shape (n_samples, n_targets)
+            The output array for the model. Rows will be removed.
+        """
+        X, y = self.mask_data(X, y, ixs=self.ixs)
+        self.est.fit(X, y)
+        return self
+
+    def predict(self, X):
+        """Call self.est.predict.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Rows will NOT be removed.
+        """
+        X = self.mask_data(X, ixs=self.ixs_pred)
+        return self.est.predict(X)
+
+    def mask_data(self, X, y=None, ixs=None):
+        """Remove datapoints according to indices or masked values.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Rows will be removed.
+        y : array, shape (n_samples, n_targets)
+            The output array for the model. Rows will be removed.
+        ixs : array, shape (n_samples_to_keep,)
+            An optional array of indices to keep before removing
+            any samples based on their values.
+
+        Returns
+        -------
+        X : array, shape (n_samples_after_removal, n_features)
+            The input X array after removing rows based on ixs, then
+            based on the values in X.
+        y : array, shape (n_samples_after_removal, n_targets) (optional)
+            The output y array after removing rows based on ixs, then
+            based on the values in X. Only returned if `y` is passed.
+        """
+
+        # First remove indices if we've manually specified them
+        if y is not None:
+            if y.ndim == 1:
+                y = y[:, np.newaxis]
```

Comment:
ya you're right, I dunno where I got that idea from...thought I remembered it throwing a deprecation warning